Misc/NEWS.d/next/Library/2018-07-08-18-49-41.bpo-33967.lhaAez.rst

```diff
@@ -0,0 +1,2 @@
+Improve functools.singledispatch exception messages when calling without
+arguments
```

Comment:
Error message is not important here. (We won't backport such improvements)

```
functools.singledispatch now raises TypeError instead of IndexError when no
positional arguments are passed.
```